sdk/openai/Azure.AI.OpenAI/api/Azure.AI.OpenAI.netstandard2.0.cs

```diff
@@ -3,22 +3,23 @@ namespace Azure.AI.OpenAI
     public partial class ChatChoice
     {
         internal ChatChoice() { }
-        public string FinishReason { get { throw null; } }
-        public int? Index { get { throw null; } }
+        public Azure.AI.OpenAI.CompletionsFinishReason FinishReason { get { throw null; } }
```

Comment:
We talked offline about customizing the constructors of the other classes that have nullable properties to manually remove the `Argument.AssertNotNull` check as a workaround for the following issue:
🔗 https://github.com/Azure/autorest.csharp/issues/2340

I'm noticing that this one here is a little bit of a special case because it involves a nullable extensible enum property.

1. Here, the `FinishReason` property is not defined as nullable (i.e., `CompletionsFinishReason? FinishReason`). This is important in this case because `CompletionsFinishReason` is a struct (i.e., a value type).

2. The generated `DeserializeChatCompletions` method deserializes the `FinishReason` property like this:
```csharp
if (property.NameEquals("finish_reason"u8)) 
{
    finishReason = new CompletionsFinishReason(property.Value.GetString());
    continue;
}
```
When the service returns null, I believe this would throw a `NullReferenceException`? But even if it didn't, this code would new up a `CompletionsFinishReason` object, and while its internal value would be null, what we really want is for there to be no object at all; `FinishReason` should be set to null.

In other words, the workaround in this case requires customizing this property and customizing this class's serialization code. Having a test for this would also be useful; I'm thinking maybe something simple like mocking a response from the service where this is null and verifying that everything works as expected.

I have added all this information to the existing issue in the autorest.csharp repo that I linked above.